sdk/identity-v2/azure-identity/src/main/java/com/azure/v2/identity/implementation/client/PersistentTokenCacheImpl.java

```diff
@@ -60,9 +61,10 @@ boolean registerCache() {
             PersistenceSettings persistenceSettings = getPersistenceSettings();
             cacheAccessAspect = new PersistenceTokenCacheAccessAspect(persistenceSettings);
             return true;
-        } catch (Throwable t) {
```

Comment:
This layer, access Native APIs via JNA.
We do get Throwables here, as error, IIRC, across platforms, e.g. if issues encounter accessing Linux Keyring etc.

Reply:
Java compiler requires checked exceptions to be caught or declared as such in the method signature. 
This code compiling means that all the checked exceptions that can be thrown are caught here. All unchecked exceptions inherit from RuntimeException, so catching IO and Runtime is enough to capture all cases.

If underlying method signature changes and a new checked exception is declared, this will be API and runtime-breaking change and this code will stop compiling forcing us to decide how to handle new exception.

JNA exceptions (at least [`LastErrorException`](https://github.com/java-native-access/jna/blob/master/src/com/sun/jna/LastErrorException.java) extends runtime one)